src/core/desktopfilewriter.cpp

```diff
@@ -0,0 +1,101 @@
+// system includes
+#include <fstream>
+#include <sstream>
+
+// local headers
+#include "linuxdeploy/util/util.h"
+#include "desktopfilewriter.h"
+
+namespace bf = boost::filesystem;
+
+namespace linuxdeploy {
+    namespace core {
+        namespace desktopfile {
+            class DesktopFileWriter::PrivateData {
+            public:
+                DesktopFile::sections_t data;
```

Comment:
It's not a good idea to expose the class data members.

Reply:
In a data class that's okay and intentional. You should check that pattern.